### ISSUE TITLE
Padding at the end of scaffolds

### ISSUE THREAD

**Issue by gbdias** [malonge/RaGOO#12]
Hi Michael,

- I am using RaGOO to scaffold a very contiguous assembly. In fact, most chromosomes are present as a single contig in my assembly, so all RaGOO should do in these cases is orient and assign them to the correct reference chromosome.
- I am curious why does RaGOO add gap padding to the end of these "single-contig scaffolds". It looks like this:
```
(...)
ACTTCTTGGTGTACGGATGTCTAACTTCTTGGTGTACGGATGTCTAACTTCTTGGTGTACGGNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN
```
- These sequence records end in a 100 bp gap despite nothing else being on the other side of the gap.
- I can easily remove this terminal gap in post-processing but I was curious if this had any purpose.

Thanks,
Guilherme

P.S. RaGOO is very useful and efficient. Great work 👍

**Comment by malonge** [comment]
Hi Guilherme,

I am glad you are finding the tool useful. Stay tuned for updates and improvements, as I am actively working on this project. 

I see what is causing this: 

https://github.com/malonge/RaGOO/blob/44068ab5726d7cc8f1c84618070750843cace1ba/ragoo.py#L243-L245

To be honest, I don't know why I added this last pad. It is reasonable given a more fragmented assembly, but you are correct that it does seem silly in this case. Give me some time to try to remember if there was a reason why I did this and I will get back to you soon with a solution. At a minimum, I could just add a command line option that disables this.

Thanks,
Mike

**Comment by gbdias** [comment]
- Cool. I'll stay tuned.
- I only noticed this because the length of my contigs increased by 100 even when no joining occurred.
- And yeah, this shouldn't matter most of the time, but when submitting the resulting scaffolds to GenBank I think it'd probably be best to remove the terminal N's first? (Not sure they would care, to be honest)

Thanks!

**Comment by malonge** [comment]
To your third point, I think it is up to your discretion. For example, if you want to place terminal gaps to indicate that there is telomere sequence not present in your assembly, that is reasonable. But not placing N's is reasonable too.

I think what I will do is add a command line option that can include or exclude terminal padding.

**Comment by malonge** [comment]
Hi there,

This has been addressed in the update to v1.1:

d6f68368addbff63b853915c00b881a46f96d35d 

I am just not adding padding to the ends of sequences. Perhaps, if people want the option in the future, I can add that functionality. But for now, it just wont add padding to the ends of sequences.

Thanks for bringing this to my attention
